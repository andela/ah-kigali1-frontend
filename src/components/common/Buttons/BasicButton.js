import React from "react";
import PropTypes from "prop-types";
const BasicButton = props => {
  return (
    <button
      type="button"
      className={`btn ${props.className}`}
      {...props}
      onClick={props.onClick}
    >
      {props.title}
    </button>
  );
};

BasicButton.prototype = {
  title: PropTypes.string,
  onClick: PropTypes.func,
  className: PropTypes.string
};
BasicButton.defaultProps = {
  title: ""
};
export default BasicButton;
