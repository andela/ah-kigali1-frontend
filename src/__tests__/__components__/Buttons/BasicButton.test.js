import React from "react";
import { shallow } from "enzyme";
import BasicButton from "../../../components/common/Buttons/BasicButton";

const setUp = (prop = {}) => {
  return shallow(<BasicButton {...prop} />);
};
const props = {
  className: "btn btn-lock",
  title: "Submit",
  onClick: jest.fn()
};
describe("Basic button", () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });
  test("should render button", () => {
    expect(component.props().title).toEqual(props.title);
    expect(component.hasClass(props.className)).toBe(true);
    expect(component.props().onClick).toBeDefined();
    component.simulate("click");
    expect(component.props().onClick).toHaveBeenCalledTimes(1);
  });
});
