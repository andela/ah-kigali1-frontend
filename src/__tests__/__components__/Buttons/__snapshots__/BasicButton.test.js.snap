// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic button should match the snapshot 1`] = `
<button
  className="btn btn-lock"
  onClick={[MockFunction]}
  title="Submit"
  type="button"
>
  Submit
</button>
`;
